Improvements for contributors

lijspoop opened the pull request:
**_No functional changes_.** Simplifies Valheim startup via configuration `Debug Run Valheim` or `Release Run Valheim`

* Update `ValheimRAFT/README.md`
  - Add a link about _KNOWNFOLDERID_
* Update `CONTRIBUTING.md`
  - Add additional properties to `.props`
  - Add some links for details
* Update `valheim.targets`, `gameLauncher.targets` and `ValheimRAFT.csproj`, `ValheimVehicles.csproj`
  - Remove duplicate properties and move them to `.props` (see `CONTRIBUTING.md`)
  - Move method of finding `GamePath` property to `gameLauncher.targets`
  
And other meaningless changes

zolantris commented:
Thanks for contributing! I'll have to test this when I get a chance.

-----

One area that would need to be kept is the dual launch of client and server. It was a solution so it's simple to swap between modes.

Run valheim should be only for the game client. But client server should start both client and server instances.

If that's kept, these changes should be fine. A bit hard to see just evaluating the xml here. I'll have to test this with rider.

Also I have a `logValheim` ps1 script I should port here. One of which does logging properly for bepinex and highlights only raft logs.

lijspoop commented:
I accidentally created a query in the wrong place, I hope everything is ok...

In general, I have implemented everything that you listed above (probably), now you can run configurations of either a dedicated server or a client, or both a dedicated server and a client.
And any of these configurations runs the logValheim script for ps1. I've changed it a bit so that you can open logs for a dedicated server by specifying the path to the log file. All you have to do is test it. 

I hope I understood correctly:
> (Although the environments should only be for debugging and release) for all solutions.

![image](https://github.com/user-attachments/assets/3c04e041-26d9-4b9a-bc37-5ebff8dfa139)
![image](https://github.com/user-attachments/assets/2fab779d-3634-4041-9033-f1bcb9286547)
etc...

lijspoop commented on ValheimMods.sln:
```diff
@@ -17,94 +17,108 @@ Global
-		Debug Run Valheim Server|Any CPU = Debug Run Valheim Server|Any CPU
-		Debug ClientServer|Any CPU = Debug ClientServer|Any CPU
 		Test|Any CPU = Test|Any CPU
+		Debug Run Client|Any CPU = Debug Run Client|Any CPU
+		Debug Run Server|Any CPU = Debug Run Server|Any CPU
+		Release Run Server|Any CPU = Release Run Server|Any CPU
+		Release Run Client|Any CPU = Release Run Client|Any CPU
+		Release Run ClientServer|Any CPU = Release Run ClientServer|Any CPU
+		Debug Run ClientServer|Any CPU = Debug Run ClientServer|Any CPU
```
Why not make a Release|Run Client instead of AnyCPU for Release Run Client|AnyCPU, to run the client and/or server, because in any case AnyCPU is used? (of course this is also for Debug) I think this will make it clearer and more convenient to switch between configurations.

FrederickEngelhardt replied:
No reason outside of not understanding what things were doing. 

I was under the assumption that any cpu was some sort of platform requirement.

This was generated from a default library template. And my c# dot net knowledge is minimal for build variants / structure.

zolantris replied:
No reason outside of not understanding what things were doing. 

I was under the assumption that any cpu was some sort of platform requirement.

This was generated from a default library template. And my c# dot net knowledge is minimal for build variants / structure.

zolantris replied:
Like the idea though. Typically do this with other npm projects. I have variant:target structure so all the releases and debug variants are next to each other.

lijspoop reviewed (commented):
I suggest replacing AnyCPU in some places of the configuration. And also use SDK tools in all csproj for better build control.

zolantris commented:
Looks good. Merged the changes.

Need to do a follow-up commit I'll make a few changes and CC you on the PR. 

The project uses doorstop `4.x.x` which allows step debugging so the script `--doorstop-target` flag must be turned into `--doorstop-target-assembly`. Also a bit on the fence where Preload should come from might be better to point it to the plugin folder similar to how the server uses the bepinex preloader references.
